src/dune/scheduler.ml

```diff
@@ -709,17 +709,11 @@ end = struct
     let fiber =
       Fiber.Var.set t_var t (fun () -> Fiber.with_error_handler f ~on_error)
     in
-    match
-      Fiber.run
-        (let* user_action_result = Fiber.fork (fun () -> fiber) in
-         let* pump_events_result = pump_events t in
-         let* user_action_result = Fiber.Future.peek user_action_result in
-         Fiber.return (pump_events_result, user_action_result))
-    with
-    | None -> Code_error.raise "[Scheduler.pump_events] got stuck somehow" []
+    match Fiber.run2 (fun () -> fiber) (fun () -> pump_events t) with
+    | _, None -> Code_error.raise "[Scheduler.pump_events] got stuck somehow" []
     | exception exn -> Error (Exn (exn, Printexc.get_raw_backtrace ()))
-    | Some (a, b) -> (
-      match (a, b) with
+    | a, Some b -> (
+      match (b, a) with
```

Comment:
This was a bit confusing until I understood the order of `a`, `b` was reversed in this `match`. I'd stick to the original order unless there's a good reason for reversing it.

Reply:
Do you mean reverse the tuples in the match branches or reverse the order of arguments of `run2`?

I feel like the match is better this way, and we can't reverse the order of argument of `run2` (because the scheduler would return immediately)

Reply:
I meant this:

> reverse the tuples in the match branches

It's not a big deal, just seemed awkward